src/modules/ejector/ejector.py

```diff
@@ -0,0 +1,198 @@
+        for no_index, validator in exit_order:
+            withdrawal_epoch = self._get_withdrawal_epoch_for_latest_validator(blockstamp, len(validators_to_eject) + 1)
+            future_rewards = (blockstamp.ref_slot - (withdrawal_epoch + epochs_to_sweep)) * rewards_speed_per_epoch
+
+            if future_rewards + total_current_balance + validator_to_eject_balance_sum >= to_withdraw_amount:
+                return validators_to_eject
+
+            validators_to_eject.append((no_index, validator))
+            validator_to_eject_balance_sum += int(validator.validator.validator.effective_balance)
+
+        return validators_to_eject
+
+    @lru_cache(maxsize=1)
+    def _get_total_balance(self, blockstamp: BlockStamp) -> Wei:
+        return Wei(
+            self._get_withdrawal_balance(blockstamp) +
+            self._get_el_vault_balance(blockstamp) +
+            self._get_reserved_buffer(blockstamp)
+        )
+
+    def _get_withdrawal_balance(self, blockstamp: BlockStamp) -> Wei:
+        return Wei(self.w3.eth.get_balance(
+            self.w3.lido_contracts.lido_locator.functions.withdrawalVault().call(
+                block_identifier=blockstamp.block_hash
+            ),
+            block_identifier=blockstamp.block_hash,
+        ))
+
+    def _get_el_vault_balance(self, blockstamp: BlockStamp) -> Wei:
+        return Wei(self.w3.eth.get_balance(
+            self.w3.lido_contracts.lido_locator.functions.elRewardsVault().call(
+                block_identifier=blockstamp.block_hash
+            ),
+            block_identifier=blockstamp.block_hash,
+        ))
+
+    def _get_reserved_buffer(self, blockstamp: BlockStamp) -> Wei:
+        return Wei(
+            self.w3.lido_contracts.lido.functions.getBufferedEther().call(
+                block_identifier=blockstamp.block_hash
+            )
+        )
+
+    def get_total_unfinalized_withdrawal_requests_amount(self, blockstamp: BlockStamp) -> Wei:
+        steth_to_finalize = self.w3.lido_contracts.withdrawal_queue_nft.functions.unfinalizedStETH().call(
+            block_identifier=blockstamp.block_hash,
+        )
+        logger.info({'msg': 'Wei to finalize.'})
+        return steth_to_finalize
+
+    def _get_withdrawal_epoch_for_latest_validator(self, blockstamp: BlockStamp, validators_to_eject_count: int) -> EpochNumber:
+        max_exit_epoch_number = 0
+        latest_to_exit_validators_count = 0
+
+        for validator in self.w3.cc.get_validators(blockstamp.state_root):
+            val_exit_epoch = EpochNumber(int(validator.validator.exit_epoch))
+
+            if val_exit_epoch == FAR_FUTURE_EPOCH:
+                continue
+
+            if val_exit_epoch == max_exit_epoch_number:
+                latest_to_exit_validators_count += 1
+
+            elif val_exit_epoch > max_exit_epoch_number:
+                max_exit_epoch_number = val_exit_epoch
+                latest_to_exit_validators_count = 1
+
+        churn_limit = self._get_churn_limit(blockstamp)
```

Comment:
we call this one too often maybe
because this method is called in the context of `for loop` inside `get_validators_to_eject`
therefore, we need to iterate over all validators each time